packages/wonder-blocks-dropdown/components/dropdown.md

```diff
@@ -203,3 +203,113 @@ class MixedDropdownExample extends React.Component {
+import Button from "@khanacademy/wonder-blocks-button";
+import Clickable from "@khanacademy/wonder-blocks-clickable";
+import {View} from "@khanacademy/wonder-blocks-core";
+import {Strut} from "@khanacademy/wonder-blocks-layout";
+import Spacing from "@khanacademy/wonder-blocks-spacing";
+import {HeadingSmall} from "@khanacademy/wonder-blocks-typography";
+import {StyleSheet} from "aphrodite";
+
+const styles = StyleSheet.create({
+    focused: {
+        border: "none",
+    },
+    hovered: {
+        textDecoration: "underline",
+    },
+    cursor: {
+        cursor: "pointer",
+        outline: "none",
+    },
+    row: {
+        flexDirection: "row",
+    }
+});
+
+class ControlledPopoverExample extends React.Component {
+    constructor() {
+        super();
+        this.state = {
+            opened: false,
+            selectedValues: ["kumail"],
+        };
+        this.handleChange = this.handleChange.bind(this);
+        this.handleClose = this.handleClose.bind(this);
+        this.openDropdown = this.openDropdown.bind(this);
+    }
+
+    handleChange(update) {
+        this.setState({
+            selectedValues: update,
+        });
+    }
+
+    handleClose() {
+        this.setState({
+            opened: false,
+        });
+    }
+
+    openDropdown() {
+        this.setState({opened: true});
+    }
+
+    render() {
+        const dropdownItems = [
+            <ActionItem label="Add new +" href="#!/Dropdown/7?new" />,
+            <SeparatorItem />,
+            <OptionItem label="Alex" value="alex" />,
+            <OptionItem label="Cathy" value="cathy" />,
+            <OptionItem label="Kumail" value="kumail" />,
+            <OptionItem label="Salman" value="salman" />,
+            <OptionItem label="Yan" value="yan" />,
+            <OptionItem label="Yash" value="yash" />,
+        ];
+
+        return (
+            <View style={styles.row}>
+                <Dropdown
+                    selectionType={"single"}
+                    menuItems={dropdownItems}
+                    onChange={this.handleChange}
+                    onClose={this.handleClose}
+                    opened={this.state.opened}
+                    selectedValues={this.state.selectedValues}
+                >
+                    {(eventState) => (
+                        <HeadingSmall
+                            onClick={() => this.setState({opened: !this.state.opened})}
+                            style={[
+                                styles.cursor,
+                                eventState.focused && styles.focused,
+                                eventState.hovered && styles.hovered,
+                            ]}
+                        >
+                            Open custom dropdown
+                        </HeadingSmall>
+                    )}
+                </Dropdown>
+                <Strut size={Spacing.medium} />
+                <Button onClick={this.openDropdown}>Open dropdown programatically</Button>
```

Comment:
After playing around with this I think it'd be nice if this toggled the dropdown open/closed.

Also, I was surprised that I was able to close the dropdown by clicking out of it - if it's a controlled component wouldn't that be entirely up to the `ControlledPopoverExample` to manage?

Reply:
Good call about the toggle fn.

And about the dropdown being closed when clicking outside, it's being managed inside the `ControlledPopoverExample` by using the `onClose` prop. If you take a look into the `handleClose` function, there's where I'm controlling the Dropdown to be hidden. Maybe the documentation needs to be more explicit about the `onClose` usage?

Reply:
Ohh, you're right - I missed that, sorry!

Reply:
But you're right, if you missed that, it could happen that the usage might be a bit confusing :). I'm going to improve this documentation in my upcoming PR (adding the controlled mode to the other dropdowns). Thanks for bringing that up!